Content/Items/Dungeon/Weapons.ThousandthDegree.cs

```diff
@@ -0,0 +1,698 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using StarlightRiver.Core;
+using StarlightRiver.Helpers;
+using StarlightRiver.Content.Items.SpaceEvent;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.Graphics.Effects;
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.DataStructures;
+using Terraria.Audio;
+
+namespace StarlightRiver.Content.Items.Dungeon
+{
+    public class ThousandthDegree : ModItem
+    {
+        public override string Texture => AssetDirectory.DungeonItem + Name;
+
+        public override bool CanUseItem(Player player) => player.ownedProjectileCounts[ModContent.ProjectileType<ThousandthDegreeProjectile>()] <= 0;
+
+        public override void SetStaticDefaults()
+        {
+            //probably a bad tooltip idk
+            Tooltip.SetDefault("Melts through enemies to build up heat\nReleasing launches the blazing wheel, melting through enemies and sliding on tiles before returning\n'Rip and tear'");
+        }
+
+        public override void SetDefaults()
+        {
+            Item.shoot = ModContent.ProjectileType<ThousandthDegreeProjectile>();
+            Item.shootSpeed = 2f;
+            Item.useStyle = ItemUseStyleID.Shoot;
+
+            Item.DamageType = DamageClass.Melee;
+            Item.damage = 42;
+            Item.knockBack = 2.5f;
+
+            Item.width = 80;
+            Item.height = 50;
+            Item.useTime = 40;
+            Item.useAnimation = 40;
+
+            Item.value = Item.sellPrice(gold: 3, silver: 50);
+            Item.rare = ItemRarityID.Orange;
+
+            Item.noMelee = true;
+            Item.autoReuse = false;
+            Item.channel = true;
+            Item.noUseGraphic = true;
+        }
+
+        public override void AddRecipes()
+        {
+            CreateRecipe().
+                AddIngredient(ModContent.ItemType<SteampunkSet.Buzzsaw>()).
+                AddIngredient(ItemID.HellstoneBar, 12).
+                AddIngredient(ItemID.Bone, 35).
+                AddTile(TileID.Anvils).
+                Register();
+        }
+    }
+
+    class ThousandthDegreeProjectile : ModProjectile
+    {
+        private const int MAXHEAT = 30;
+
+        private int flashTimer;
+
+        private float wheelRot;
+
+        private bool fired;
+
+        private bool flashed;
+
+        private Projectile fireProjectile;
+
+        public ref float CurrentHeat => ref Projectile.ai[0];
```

Comment:
It would probably be wise to move the charge ownership to the item to be consistent with other charged items. While this is slightly easier to sync, its difficult to make consistent with charged items that dont use a holdout projectile.